### DISCUSSION TITLE
xmake config should not cache the unsuccessful result when finding tools

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#1705] [General]
> 注：如果是`问题报障`或者`特性请求`，请选择`报告缺陷`，`特性请求`模板，否则一律不回复。

### 描述讨论详情

```
> xmake
error: swig not found!
> (set up environment)
> xmake
error: swig not found!
> xmake f -c
> xmake
(successfully build)
```

当find_tool失败的时候xmake会缓存失败的结果，再次运行时不会重新探测；但既然失败了下次运行肯定是尝试修复之后才运行，所以不应该缓存失败的结果

### 期望的行为

```
> xmake
error: swig not found!
> (set up environment)
> xmake
(successfully build)
```

**Comment by waruqi** [comment]
缓存失败的目的就是为了下次不再重新探测，否则每次执行 如果有十几个可选tool都一直不存在，那就要每次重新检测。

不仅影响效率 而且 xmake f -c 也就没意义了

毕竟用户的环境不可能一直在变，100次运行，而用户安装上此工具只变动一次环境，甚至可能几天几月才有变动，却要每次都去检测直到成功为止

还不如失败后缓存，之后每次都能快速执行，如果用户环境变动 装上了对应工具自己手动 xmake f -c 强制检测下就完事了

**Comment by waruqi** [comment]
这个目前就是这么设计的 自己 xmake f -c就好了 暂时不考虑做改动
